openedx/core/djangoapps/user_api/helpers.py

```diff
@@ -407,26 +407,32 @@ def shim_student_view(view_func, check_logged_in=False):
     """
     @wraps(view_func)
     def _inner(request):  # pylint: disable=missing-docstring
-        # Ensure that the POST querydict is mutable
-        request.POST = request.POST.copy()
+        # Make a copy of the current POST request to modify.
+        modified_request = request.POST.copy()
+        if isinstance(request, HttpRequest):
+            # Works for an HttpRequest but not a rest_framework.request.Request.
+            request.POST = modified_request
+        else:
+            # The request must be a rest_framework.request.Request.
+            request._data = modified_request
```

Comment:
A request serviced by `shim_student_view()` can be either an `HttpRequest` -or- a DRF `Request`. The objects can no longer be handled the same way. I'm curious whether this whole pattern of modifying the request can be scrapped in favor of something else.